hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/UnknownStoreException.java

```diff
@@ -33,23 +32,28 @@
  */
 @InterfaceAudience.Public
 @InterfaceStability.Evolving
-public class UnknownStoreException extends IOException {
+public class UnknownStoreException extends PathIOException {
 
   /**
    * Constructor.
-   * @param message message
+   *
+   * @param message message.
+   * @param path    path trying to access.
    */
-  public UnknownStoreException(final String message) {
-    this(message, null);
+  public UnknownStoreException(final String message, final String path) {
```

Comment:
can you move path to be the first argument, to be consistent with PathIOE itself?